cfme/utils/config.py

```diff
@@ -0,0 +1,141 @@
+        return self.config
+
+    def __setitem__(self, key, value):
+        """This function allows you to mimic some behavior like runtime from yaycl
+            You can do following:
+                from cfme.utils import conf
+                conf ['env']['new_key'] = 'new_val'
+                conf ['env']['new_key']
+                # The new_key and its value is cached through the session
+        """
+        self.settings[key] = value
+
+    def __contains__(self, key):
+        return key in self.settings.as_dict() or key.upper() in self.settings.as_dict()
+
+    def get(self, key, default=None):
+        try:
+            return self.__getattr__(key)
+        except KeyError:
+            return default
+
+
+class Configuration(object):
+    """
+    Holds ConfigData objects in dictionary `settings`.
+    """
+    def __init__(self):
+        self.settings = None
+        self.yaycl_config = None
+
+    def configure_creds(self, config_dir, crypt_key_file=None):
+        """
+        do the defered initial loading of the configuration
+
+        :param config_dir: path to the folder with configuration files
+        :param crypt_key_file: optional name of a file holding the key for encrypted
+            configuration files
+
+        :raises: AssertionError if called more than once
+
+        if the `utils.conf` api is removed, the loading can be transformed to eager loading
+        """
+        assert self.yaycl_config is None
+        if crypt_key_file and os.path.exists(crypt_key_file):
+            self.yaycl_config = yaycl.Config(
+                config_dir=config_dir,
+                crypt_key_file=crypt_key_file)
+        else:
+            self.yaycl_config = yaycl.Config(config_dir=config_dir)
+
+    def configure(self):
+        """
+        Loads all the congfigs in settings dictionary where Key is filename(without extension)
+        and value is ConfigData object relevant to that file.
+
+        Example self.settings would look like:
+        {'cfme_data': <cfme.utils.config_copy.ConfigData at 0x7f2a943b14e0>,
+         'env': <cfme.utils.config_copy.ConfigData at 0x7f2a943b1be0>,
+         'gpg': <cfme.utils.config_copy.ConfigData at 0x7f2a943b1cc0>,
+         ...
+         ...
+         'polarion_tools': <cfme.utils.config_copy.ConfigData at 0x7f2a943ad438>,
+         'perf_tests': <cfme.utils.config_copy.ConfigData at 0x7f2a943ad2e8>}
+
+        """
+        if self.settings is None:
+            # -5 means removing `.yaml` from filename and just keeping basename
+            self.settings = {
+                file.basename[:-5]: ConfigData(env=file.basename[:-5])
+                for file in path.conf_path.listdir() if file.basename[-5:] == '.yaml' and
+                '.local.yaml' not in file.basename
+            }
+
+    def get_config(self, name):
+        """returns a config object
+
+        :param name: name of the configuration object
+        """
+        self.configure()
```

Comment:
No need to explicitly call this when the data that its filling is provided by the cached_property on access.

Reply:
removed